### PR TITLE
FEAT(#007): ADD MATERIAL DETAILS PAGE

### PR DESCRIPTION
**Pull request by ivamorgra**
# Pull Request Description

## Changes Made

I have implemented a new feature in the `MaterialDetailsPage` component and added a new `FileComponent` to enhance the user experience in viewing and downloading course materials.

### MaterialDetailsPage Component

- [x] **Code Structure and Readability**
  - [x] The code follows a clear and consistent structure.
  - [x] Variable and function names are descriptive and adhere to coding conventions.
  - [x] There are no unnecessary or redundant code blocks.

- [x] **Component Logic**
  - [x] The `useEffect` hook fetches material details based on the provided ID.
  - [x] The `FileComponent` is correctly imported and integrated into the `MaterialDetailsPage`.
  - [x] The 'Download' button triggers the `handleDownload` function, downloading the correct file.

- [x] **Responsiveness**
  - [x] The component adjusts appropriately to different screen sizes, utilizing the `useResponsive` hook.

### FileComponent

- [x] **Code Structure and Readability**
  - [x] The code follows a clear and consistent structure.
  - [x] Variable and function names are descriptive and adhere to coding conventions.
  - [x] There are no unnecessary or redundant code blocks.

- [x] **File Rendering**
  - [x] The component correctly renders the title, description, and an `iframe` for the provided file URL.
  - [x] The `iframe` has appropriate attributes (width, height, title) for a PDF viewer.

- [x] **Styling**
  - [x] The card, typography, and divider components are styled appropriately.
  - [x] The background color and styling of the description text follow the design specifications.

Thank you for your time and attention to this pull request!

### CODE REVIEW

File: client/src/components/course-classes-materials/course-materials.tsx [hunk 1]
```diff
@@ -41,49 +41,54 @@ export default function CourseMaterials({
...
-                                <IconButton>
-                                    <ShoppingCart />
-                                </IconButton>
-                            )}
-                        </ListItemIcon>
-                    </CardContent>
-                </Card>
-            ))}
+          {materials.map((material) => (
+            // Utiliza Link para envolver el área clickeable
```
Comment (mrarjona) [inline client/src/components/course-classes-materials/course-materials.tsx]: You can delete this unnecessary comment

File: client/src/pages/material-details.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+// material-details.tsx
```
Comment (mrarjona) [inline client/src/pages/material-details.tsx:1]: You can delete this unnecessary comment

File: client/tsconfig.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
-    "jsx": "react-jsx"
+    "jsx": "preserve"
```
Comment (mrarjona) [inline client/tsconfig.json]: Can this change cause any problems for other project teams?

File: client/src/pages/material-details.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        setMaterialDetails(foundMaterial);
+      }
+    };
+
+    getMaterialDetails();
+  }, [authUser, id]);
+
+    const isSmallScreen = useResponsive('down', 'sm')
+    const handleDownload = () => {
+      const link = document.createElement('a');
```
Comment (mrarjona) [inline client/src/pages/material-details.tsx:36]: Could you add the option `link.target = '_blank';` to open a new tab when clicking on download?

File: client/tsconfig.json (hunk 3 above)
Comment (ivamorgra) [reply client/tsconfig.json]: I think "react-jsx" mode, TypeScript compiles JSX to React.createElement calls. It is the default setting for React projects with TypeScript.

By another hand, With "preserve" setting, TypeScript doesn't transform JSX and keeps it unchanged in the compiled code.

I have restored this configuration, I dont know how I have changed this haha.

Thank you for your review.

**Review by mrarjona — DISMISSED** [review]
All perfect @ivamorgra, thanks for taking into account the suggestions! 🤝

**Review by mrarjona — APPROVED** [review]

**Pull request merged**